core/src/main/java/hudson/FilePath.java

```diff
@@ -1829,7 +1830,12 @@ public boolean accept(File f) {
             throw new IOException("Expecting Ant GLOB pattern, but saw '"+includes+"'. See http://ant.apache.org/manual/Types/fileset.html for syntax");
         FileSet fs = Util.createFileSet(dir,includes,excludes);
         fs.setDefaultexcludes(defaultExcludes);
-        DirectoryScanner ds = fs.getDirectoryScanner(new Project());
+        DirectoryScanner ds;
+        try {
+            ds = fs.getDirectoryScanner(new Project());
+        } catch (BuildException x) {
+            throw new IOException(x.getMessage());
```

Comment:
Such approach causes data loss, because you do not propagate Location from `BuildException`. This is the main difference from #3358 in the implementation PoV.

Reply:
The places in `getDirectoryScanner` which construct a `BuildException` do not provide a `Location`—this class is only used for actual build failures during an Ant build—so there is no data loss AFAICT.